src/app/initialisation/run.js

```diff
@@ -57,42 +59,45 @@ function initCoreModules(mediator) {
  * Monitoring any changes in the state to verify that the user
  * @param $rootScope
  * @param $state
- * @param $q
- * @param mediator
  * @param userClient
  */
-function verifyLoginOnStateChange($rootScope, $state, $q, mediator, userClient) {
-  var initPromises = [];
-  var initListener = mediator.subscribe('promise:init', function(promise) {
-    initPromises.push(promise);
-  });
-  mediator.publish('init');
-  var all = (initPromises.length > 0) ? $q.all(initPromises) : $q.when(null);
-  all.then(function() {
-    $rootScope.ready = true;
-    mediator.remove('promise:init', initListener.id);
-    return null;
-  });
+function verifyLoginOnStateChange($rootScope, $state, userClient) {
+
+  $rootScope.ready = false;
+
+  function clearAndRedirectToLogin() {
+    userClient.clearSession().then(function() {
+      $state.go('app.login');
+    });
+  }
 
   $rootScope.$on('$stateChangeStart', function(e, toState, toParams) {
 
-    function clearAndRedirectToLogin() {
-      userClient.clearSession().then(function() {
-        $rootScope.toState = toState;
-        $rootScope.toParams = toParams;
-        $state.go('app.login');
-      });
+    //If the session is valid, or accessing the login page, then the transition is valid
+    if (isAuthorised || toState.name === "app.login") {
```

Comment:
maybe it would be nice to move the list of states acessible without auth to a more prominent place, even if it's not likely to change.